### PR TITLE
smart BFT GRPC leader sends proposal to follower test

### PR DESCRIPTION
**Pull request by semil**
SmartBFT integration test:
The leader is using GRPC to send a proposal to the follower, checking that the signature validation is correct.

#### Type of change

- Test update

### CODE REVIEW

File: integration/smartbft/smartbft_test.go [hunk 1]
```diff
@@ -1398,7 +1401,7 @@ var _ = Describe("EndToEnd Smart BFT configuration test", func() {
 			}
 
 			By("Expect an error")
-			Expect(client.Send(req)).NotTo(HaveOccurred())
+			Expect(client.Send(req)).To(Succeed())
```
Comment (yacovm) [inline integration/smartbft/smartbft_test.go:1404]: why this change?

File: integration/smartbft/smartbft_test.go [hunk 2]
```diff
@@ -1415,11 +1418,189 @@ var _ = Describe("EndToEnd Smart BFT configuration test", func() {
...
+			genesisBlock := network.LoadAppChannelGenesisBlock(channel)
+			req, _ := createPrePrepareRequest(genesisBlock, genesisBlock, env, channel)
+
+			By("Send the request")
+			Expect(client.Send(req)).To(Succeed())
+
+			<-time.After(3 * time.Second)
+
+			By("Assert failure")
+			Eventually(ordererRunners[follower.Id-1].Err(), network.EventuallyTimeout).Should(gbytes.Say("SigFilter evaluation failed: implicit policy evaluation failed - 0 sub-policies were satisfied"))
```
Comment (yacovm) [inline integration/smartbft/smartbft_test.go]: this is not the message we want. we want to see there is a view change.

**Review by yacovm — COMMENTED** [review]
Can you put case II first and then there would be a view change, and then do case I from the new leader?

File: integration/smartbft/smartbft_test.go [hunk 3]
```diff
@@ -1415,11 +1418,189 @@ var _ = Describe("EndToEnd Smart BFT configuration test", func() {
...
+
+			By("Create the consensus request")
+			genesisBlock := network.LoadAppChannelGenesisBlock(channel)
+			req, _ := createPrePrepareRequest(genesisBlock, genesisBlock, env, channel)
+
+			By("Send the request")
+			Expect(client.Send(req)).To(Succeed())
+
+			By("Assert success")
+			Eventually(ordererRunners[follower.Id-1].Err(), network.EventuallyTimeout).Should(gbytes.Say("LogRecord appended successfully"))
```
Comment (yacovm) [inline integration/smartbft/smartbft_test.go]: this does not say much. can we find a more indicative message?

Comment (semil) [reply integration/smartbft/smartbft_test.go]: When node 2 sends pre-prepare to node 1 the log of 1 is: 

```
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 0205 DEBU [orderer.consensus.smartbft.chain] HandleMessage -> Message from 2 channel=testchannel1
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 0206 DEBU [orderer.consensus.smartbft.consensus] ProcessMessages -> 1 got message from 2: <PrePrepare with view: 1, seq: 1, payload of 974 bytes, header: MEcCAQEEIO2JmWQbABTpxQPAC3Nthbbu4pic+/pPTB9FaVoSKtdmBCBlv8DoUW6c1Fd78f/bD8WePESdQXiCZWzIe3TryXhcFg==> channel=testchannel1
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 0207 DEBU [orderer.consensus.smartbft.consensus] handleHeartBeat -> Received heartbeat from 2, last heartbeat was 1.999168753s ago channel=testchannel1
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 0208 DEBU [orderer.consensus.smartbft.consensus] processMsg -> 1 got message <PrePrepare with view: 1, seq: 1, payload of 974 bytes, header: MEcCAQEEIO2JmWQbABTpxQPAC3Nthbbu4pic+/pPTB9FaVoSKtdmBCBlv8DoUW6c1Fd78f/bD8WePESdQXiCZWzIe3TryXhcFg==> from 2 with seq 1 channel=testchannel1
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 0209 DEBU [policies] EvaluateSignedData -> == Evaluating *policies.ImplicitMetaPolicy Policy /Channel/Writers ==
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 020a DEBU [policies] EvaluateSignedData -> This is an implicit meta policy, it will trigger other policy evaluations, whose failures may be benign
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 020b DEBU [policies] EvaluateSignedData -> == Evaluating *policies.ImplicitMetaPolicy Policy /Channel/Application/Writers ==
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 020c DEBU [policies] EvaluateSignedData -> This is an implicit meta policy, it will trigger other policy evaluations, whose failures may be benign
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 020d DEBU [policies] EvaluateSignedData -> == Evaluating *cauthdsl.policy Policy /Channel/Application/Org1/Writers ==
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 020e DEBU [policies] SignatureSetToValidIdentities -> signature for identity 0 validated
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 020f DEBU [policies] EvaluateSignedData -> Signature set did not satisfy policy /Channel/Application/Org1/Writers
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 0210 DEBU [policies] EvaluateSignedData -> == Done Evaluating *cauthdsl.policy Policy /Channel/Application/Org1/Writers
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.341 IDT 0211 DEBU [policies] EvaluateSignedData -> == Evaluating *cauthdsl.policy Policy /Channel/Application/Org2/Writers ==
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0212 DEBU [policies] SignatureSetToValidIdentities -> signature for identity 0 validated
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0213 DEBU [policies] EvaluateSignedData -> Signature set did not satisfy policy /Channel/Application/Org2/Writers
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0214 DEBU [policies] EvaluateSignedData -> == Done Evaluating *cauthdsl.policy Policy /Channel/Application/Org2/Writers
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0215 DEBU [policies] func1 -> Evaluation Failed: Only 0 policies were satisfied, but needed 1 of [ Org1/Writers Org2/Writers ]
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0216 DEBU [policies] EvaluateSignedData -> Signature set did not satisfy policy /Channel/Application/Writers
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0217 DEBU [policies] EvaluateSignedData -> == Done Evaluating *policies.ImplicitMetaPolicy Policy /Channel/Application/Writers
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0218 DEBU [policies] EvaluateSignedData -> == Evaluating *policies.ImplicitMetaPolicy Policy /Channel/Orderer/Writers ==
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0219 DEBU [policies] EvaluateSignedData -> This is an implicit meta policy, it will trigger other policy evaluations, whose failures may be benign
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 021a DEBU [policies] EvaluateSignedData -> == Evaluating *cauthdsl.policy Policy /Channel/Orderer/OrdererOrg/Writers ==
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 021b DEBU [policies] SignatureSetToValidIdentities -> signature for identity 0 validated
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 021c DEBU [policies] EvaluateSignedData -> Signature set satisfies policy /Channel/Orderer/OrdererOrg/Writers
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 021d DEBU [policies] EvaluateSignedData -> == Done Evaluating *cauthdsl.policy Policy /Channel/Orderer/OrdererOrg/Writers
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 021e DEBU [policies] EvaluateSignedData -> Signature set satisfies policy /Channel/Orderer/Writers
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 021f DEBU [policies] EvaluateSignedData -> == Done Evaluating *policies.ImplicitMetaPolicy Policy /Channel/Orderer/Writers
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0220 DEBU [policies] EvaluateSignedData -> Signature set satisfies policy /Channel/Writers
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0221 DEBU [policies] EvaluateSignedData -> == Done Evaluating *policies.ImplicitMetaPolicy Policy /Channel/Writers
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0222 DEBU [orderer.consensus.smartbft.consensus] verifyPrevCommitSignatures -> Previous proposal verification sequence: 18446744073709551615, current verification sequence: 0 channel=testchannel1
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0223 INFO [orderer.consensus.smartbft.consensus] verifyPrevCommitSignatures -> Skipping verifying prev commit signatures due to verification sequence advancing from 18446744073709551615 to 0 channel=testchannel1
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.342 IDT 0224 DEBU [orderer.consensus.smartbft.consensus] verifyBlacklist -> DecisionsPerLeader is 0, hence leader rotation is inactive channel=testchannel1
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.363 IDT 0225 DEBU [orderer.consensus.smartbft.chain] append -> LogRecord appended successfully: total size=1160, recordLength=1148, dataCRC=E09F2D39; file=/var/folders/sh/cx2kqdf52bbgygbr8vchj1qc0000gn/T/e2e-smartbft-test842847274/orderers/OrdererOrg.orderer1/etcdraft/wal/testchannel1/0000000000000001.wal, new-offset=1208 channel=testchannel1
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.363 IDT 0226 INFO [orderer.consensus.smartbft.consensus] processProposal -> Processed proposal with seq 1 channel=testchannel1
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.363 IDT 0227 DEBU [orderer.consensus.smartbft.rpc] consensusSent -> Sending msg of 72 bytes to 2 on channel testchannel1 took 13.888µs channel=testchannel1
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.363 IDT 0228 DEBU [orderer.consensus.smartbft.rpc] consensusSent -> Sending msg of 72 bytes to 3 on channel testchannel1 took 3.362µs channel=testchannel1
[e][OrdererOrg.orderer1] 2023-09-13 16:40:30.363 IDT 0229 DEBU [orderer.consensus.smartbft.rpc] consensusSent -> Sending msg of 72 bytes to 4 on channel testchannel1 took 3.696µs channel=testchannel1
```

Comment (yacovm) [reply integration/smartbft/smartbft_test.go]:
```
Processed proposal with seq 1 channel=testchannel1
```

Maybe try this

**Review by yacovm — APPROVED** [review]
